### PR TITLE
Revision reader tweak

### PR DESCRIPTION
**Pull request by gerhardol**
## Proposed changes

Avoid replace \v in email etc (not needed, performance).

Make revision reading slightly easier to read.

[IDE0008](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008) in RevisionReader.cs

This originated as a review to see if it was possible to improve the performance, primarily to see if allocations could be eliminated, but it does not seem there is big difference.
Still, this is critical code when opening repos.

## Test methodology 

Manual

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/RevisionReader.cs [hunk 1]
```diff
@@ -290,52 +290,50 @@ int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
...
-            var committerEmail = reader.ReadLine();
-            var reflogSelector = _hasReflogSelector ? reader.ReadLine() : null;
-
-            bool skipBody = _sixMonths > authorUnixTime;
-            (string? subject, string? body, bool hasMultiLineMessage) = reader.PeekSubjectBody(skipBody);
+            string author = reader.ReadLine();
+            string authorEmail = reader.ReadLine();
+            string committer = reader.ReadLine();
+            string committerEmail = reader.ReadLine();
+            string reflogSelector = _hasReflogSelector ? reader.ReadLine(useStringPool: false) : null;
```
Comment (mstv) [inline GitCommands/RevisionReader.cs:329]: It is not obvious to me why (just) in this case the string is not shared.
Ah, the regarding comment is located inside the function. Move it here?

File: GitCommands/RevisionReader.cs [hunk 2]
```diff
@@ -403,34 +401,42 @@ public StringLineReader(in ReadOnlySpan<char> s)
...
+                    : bodySlice)
+                    .ToString();
 
                 // See caller for reasoning when message body can be omitted
                 // (String interning makes hasMultiLineMessage check only for clarity)
-                string? body = skipBody || !hasMultiLineMessage
-                    ? null
-                    : bodySlice.ToString();
+                body = keepBody && hasMultiLineMessage
+                    ? bodySlice.ToString().Replace('\v', '\n')
```
Comment (mstv) [inline GitCommands/RevisionReader.cs:436]: ❔
```suggestion
                    ? bodySlice.Replace('\v', '\n').ToString()
```

File: GitCommands/RevisionReader.cs [hunk 3]
```diff
@@ -403,34 +401,42 @@ public StringLineReader(in ReadOnlySpan<char> s)
...
+                    : null;
 
-                return (subject, body, hasMultiLineMessage);
+                return;
```
Comment (mstv) [inline GitCommands/RevisionReader.cs:439]: not necessary as last statement
```suggestion
```

**Review by mstv — COMMENTED** [review]
LGTM

File: GitCommands/RevisionReader.cs [hunk 4]
```diff
@@ -403,34 +401,42 @@ public StringLineReader(in ReadOnlySpan<char> s)
...
                 _index += lineLength + 1;
-                return StringPool.Shared.GetOrAdd(_s.Slice(startIndex, lineLength));
+
+                // Authors etc are limited, use a shared string pool
+                ReadOnlySpan<char> s = _s.Slice(startIndex, lineLength);
+                return useStringPool ? StringPool.Shared.GetOrAdd(s) : s.ToString();
             }
 
-            public (string? subject, string? body, bool hasMultiLineMessage) PeekSubjectBody(bool skipBody)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (RussKie) [inline GitCommands/RevisionReader.cs:410]: How did you decide this attribute is necessary?

**Review by RussKie — COMMENTED** [review]
👍

File: GitCommands/RevisionReader.cs (hunk 1 above)
Comment (gerhardol) [reply GitCommands/RevisionReader.cs:329]: Moving the comment, separate for authors etc and reflogselector

File: GitCommands/RevisionReader.cs (hunk 2 above)
Comment (gerhardol) [reply GitCommands/RevisionReader.cs:436]: You cannot do this on a Span<>

File: GitCommands/RevisionReader.cs (hunk 4 above)
Comment (gerhardol) [reply GitCommands/RevisionReader.cs:410]: This routine can be executed 100K every time the revisions are reloaded, so speed is preferred over code size
I considered removing the internal class StringLineReader to give the compiler some more hints.
I doubt this is measurable though.

**Review by mstv — APPROVED** [review]
👍

**Review by RussKie — APPROVED** [review]

**Pull request merged**